pkg/cmd/admin/policy/policy.go

```diff
@@ -187,17 +185,21 @@ func NewClusterRoleBindingAccessor(client client.Interface) ClusterRoleBindingAc
 }
 
 func (a ClusterRoleBindingAccessor) GetExistingRoleBindingsForRole(roleNamespace, role string) ([]*authorizationapi.RoleBinding, error) {
-	uncast, err := a.Client.ClusterPolicyBindings().Get(authorizationapi.GetPolicyBindingName(roleNamespace), metav1.GetOptions{})
-	if err != nil && !kapierrors.IsNotFound(err) {
+	// cluster role bindings can only reference cluster roles
+	if roleNamespace != "" {
+		return nil, nil
+	}
+
+	existingBindings, err := a.Client.ClusterRoleBindings().List(metav1.ListOptions{})
+	if err != nil {
 		return nil, err
 	}
-	existingBindings := authorizationapi.ToPolicyBinding(uncast)
 
 	ret := make([]*authorizationapi.RoleBinding, 0)
 	// see if we can find an existing binding that points to the role in question.
-	for _, currBinding := range existingBindings.RoleBindings {
-		if currBinding.RoleRef.Name == role {
-			t := currBinding
+	for _, currBinding := range existingBindings.Items {
+		if currBinding.RoleRef.Name == role && currBinding.RoleRef.Namespace == "" {
+			t := authorizationapi.ToRoleBinding(&currBinding)
```

Comment:
No but this client side so it should not matter (the previous code was just a shallow copy also).